api/src/main/java/org/openmrs/module/fhir/api/impl/DiagnosticReportServiceImpl.java

```diff
@@ -68,16 +93,31 @@ public void setDao(FHIRDAO dao) {
 	}
 
 	@Override
-	public DiagnosticReport getDiagnosticReport(String id) {
-		// Find Diagnostic Report (Encounter) in OpenMRS database
-		EncounterService encounterService = Context.getEncounterService();
-		Encounter omrsDiagnosticReport = encounterService.getEncounterByUuid(id);
-		// Get corresponding Handler
-		String handlerName = omrsDiagnosticReport.getEncounterType().getName();
+	public DiagnosticReport getDiagnosticReport(final String reportId) {
+		loadHandlerMap();
+		List<Order> orders = dao.getOrdersByAccessionNumber(reportId);
+		int orderId = 0;
+		if ((orders != null) && !orders.isEmpty()) {
+			orderId = orders.get(0).getOrderId();
+		}
+
+		if (orderId > 0) {
+			Order order = Context.getOrderService().getOrder(orderId);
+			if (order == null) {
+				throw new RuntimeException("Can not find order by the accession number");
+			}
+			String handlerName = orderTypeToHandlerNameMap.get(order.getOrderType().getName());
+			if (StringUtils.isEmpty(handlerName)) {
+				handlerName = orderTypeToHandlerNameMap.get("DEFAULT");
```

Comment:
Since you are using `DEFAULT` in more than one place, do you wanna have a constant for it?